### PR TITLE
Add support to convert between logs and annotations

### PR DESCRIPTION
**Pull request by worldtiki**
### :pencil: Description
Add support to convert annotations to logs.

Thanks @jcchavezs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -264,6 +265,12 @@
       <version>${awaitility.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.assertj</groupId>
+      <artifactId>assertj-core</artifactId>
+      <version>${assertj-core.version}</version>
```
Comment (worldtiki) [inline pom.xml]: this is not related to the pr but, this library is amazing. fluent + easy assertions.

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java [hunk 2]
```diff
@@ -48,12 +49,22 @@ public static Span fromZipkinV2(zipkin2.Span zipkin) {
...
+                .collect(toList());
+        builder.addAllTags(tags);
+
+        var logs = zipkin.annotations().stream()
+                .map(annotation -> Log.newBuilder()
+                        .setTimestamp(annotation.timestamp())
+                        .addFields(Tag.newBuilder()
+                                .setVStr(annotation.value())
+                                .setType(Tag.TagType.STRING)
+                                .setKey("event")
```
Comment (worldtiki) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java]: waiting on the haystack folks to confirm if "event" is a good name for the key.
this has impact on the way data is indexed so we may want to leave this null or configurable.

File: pom.xml (hunk 1 above)
Comment (jcchavezs) [reply pom.xml]: I think if we go with this library then we should move all assertions into using that (into another PR). That could be fun.

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java (hunk 2 above)
Comment (eleduardo) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java]: In theory "logs" are not indexed as they are micro-events inside the span so any key should be fine but agree on letting the Haystack folks confirm

Comment (adriancole) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java]: another way is to use the value as a key and empty string as value.. depends on how the backend works and you'll know best

File: pom.xml (hunk 1 above)
Comment (worldtiki) [reply pom.xml]: it's a small change. i'll do it immediately after this one.

File: src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java [hunk 3]
```diff
@@ -48,12 +49,21 @@ public static Span fromZipkinV2(zipkin2.Span zipkin) {
...
+        List<Tag> tags = zipkin.tags().entrySet().stream()
+                .flatMap(entry -> fromZipkinTag(entry.getKey(), entry.getValue()).stream())
+                .collect(toList());
+        builder.addAllTags(tags);
+
+        var logs = zipkin.annotations().stream()
+                .map(annotation -> Log.newBuilder()
+                        .setTimestamp(annotation.timestamp())
+                        .addFields(Tag.newBuilder()
+                                .setKey("annotation")
```
Comment (jcchavezs) [inline src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java]: @worldtiki could you please link the line in the agent where this is happening?

Comment (worldtiki) [reply src/main/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverter.java]: of course. i should have done that, sorry

https://github.com/ExpediaDotCom/haystack-agent/blob/master/agent-providers/span/src/main/java/com/expedia/www/haystack/agent/pitchfork/processors/HaystackDomainConverter.java#L65-L72

**Pull request merged**